beacon-chain/core/peerdas/helpers.go

```diff
@@ -75,28 +83,43 @@ func CustodyColumnSubnets(nodeId enode.ID, custodySubnetCount uint64) (map[uint6
 	// First, compute the subnet IDs that the node should participate in.
 	subnetIds := make(map[uint64]bool, custodySubnetCount)
 
-	for i := uint64(0); uint64(len(subnetIds)) < custodySubnetCount; i++ {
-		nodeIdUInt256, nextNodeIdUInt256 := new(uint256.Int), new(uint256.Int)
-		nodeIdUInt256.SetBytes(nodeId.Bytes())
-		nextNodeIdUInt256.Add(nodeIdUInt256, uint256.NewInt(i))
-		nextNodeIdUInt64 := nextNodeIdUInt256.Uint64()
-		nextNodeId := bytesutil.Uint64ToBytesLittleEndian(nextNodeIdUInt64)
+	// Convert the node ID to a big int.
+	nodeIdUInt256 := new(uint256.Int).SetBytes(nodeId.Bytes())
 
-		hashedNextNodeId := hash.Hash(nextNodeId)
-		subnetId := binary.LittleEndian.Uint64(hashedNextNodeId[:8]) % dataColumnSidecarSubnetCount
+	// Handle the maximum value of a uint256 case.
+	if nodeIdUInt256.Cmp(maxUint256) == 0 {
```

Comment:
Why set it to 0 ? Do spec tests expect it

Reply:
Just realized they changed the implementation, 
https://github.com/ethereum/consensus-specs/blob/dev/specs/_features/eip7594/das-core.md#get_custody_columns

Reply:
I proposed a new change: https://github.com/ethereum/consensus-specs/pull/3742